File: 10/Circulo.cpp
#include <iostream>
#include <math.h>
#include <String.h>

using namespace std;

double r, C , S;
char aux[20];
//double atof();

int isNum (char *cad){
	for (int i=0; i < strlen (cad) ; i++)
	if(cad[i] =='.' || cad[i] =='-' || cad[i] == '\r'  || cad[i] == '\n')
		continue;
	else if (cad[i] < '0' || cad[i] > '9') {
     cout << "solo se admiten numeros";
     return 1;
    }
return 0;
}

double CapReal(char *prompt){
	 do{
		cout<< prompt;
	 fgets(aux, sizeof(aux), stdin);
	 }while (isNum(aux) !=0);
	 return atof(aux);
}

void inicio() {
	cout << "Para calccular la superficie y la circunfernecia";
	cout << " de circulos cualquiera";
}

void datos() {
	float rad;
	do rad = CapReal ("\nDe valor del radio:");
	while (rad < 0);
	r = rad;
}

void calculos() {
	S = M_PI * r * r;
	C = 2 * M_PI * r;
}

void resultados() {
	cout << "\n  cuando el radio vale: " << r;
	cout << "\n la superficie es " << S;
	cout << "\n la circunferencia es " << C;
}

int main() {
	int resp = 'S';

     inicio();
     while (resp == 'S' ||resp == 's') {
	 datos();
     calculos();
	 resultados();
	 cout << "\nCalcular otro circulo? s/n:";
	 resp = getchar();
	 getchar();
	}
}

File: 8/Circulo.java
import java.io.*;
import java.util.Scanner;

public class Circulo {
    Scanner tecla = new Scanner(System.in);
    double r, c, s;

    void inicio(){
        System.out.println("Programa para calcular y\nmostrar el perimetro y la\nsuperficie circulos cualesquiera\n");
    }

    boolean esNumero(String cad){
        try {
            Double.parseDouble(cad);
            return true;
        } catch (NumberFormatException nfe) {
            System.out.print("\nNo puedo procesar el dato introducido\nPor favor ingrese un numero ... ");
        return false;
        }
    }

    void datos(){
        String aux = " ";

        do {
            do{
                System.out.print("\nIntrodusca el valor del radio ... ");
                aux = tecla.nextLine();
            } while( !esNumero(aux) );
            r = Double.parseDouble(aux);
        } while(r<0);
    }

    void calculos(){
        c = Math.PI * 2 * r;
        s = Math.PI * Math.pow(r,2);
    }

    void resultados(){
        System.out.print("\nCuando el valor del radio es:"+r+"\nLa superficie es igual a "+s+"\nY la circunferencia es "+c);
    }

public static void main(String args[]) throws IOException {
    int resp = 's';
    Circulo C = new Circulo();

    C.inicio();
    while(resp == 'S' || resp == 's'){
        C.datos();
        C.calculos();
        C.resultados();
        System.out.print("\nDesea calcular otro circulo?\nS/N ... ");
        resp = System.in.read();
        System.in.skip(2);
    }
}
}
